src/language-js/parser-meriyah.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+
+const createError = require("../common/parser-create-error");
+const { hasPragma } = require("./pragma");
+const { locStart, locEnd } = require("./loc");
+const postprocess = require("./postprocess");
+
+// https://github.com/meriyah/meriyah/blob/4676f60b6c149d7082bde2c9147f9ae2359c8075/src/parser.ts#L185
+const parseOptions = {
+  // Allow module code
+  module: true,
+  // Enable stage 3 support (ESNext)
+  next: true,
+  // Enable start and end offsets to each node
+  ranges: true,
+  // Enable web compatibility
+  webcompat: true,
+  // Enable line/column location information to each node
+  loc: true,
+  // Attach raw property to each literal and identifier node
+  raw: true,
+  // Enabled directives
+  directives: true,
+  // Allow return in the global scope
+  globalReturn: true,
+  // Enable implied strict mode
+  impliedStrict: false,
+  // Enable non-standard parenthesized expression node
+  preserveParens: false,
```

Comment:
We have this enabled for Babel to support Closure style type assertions, so let's enable it for meriyah too.

Reply:
Okay

Reply:
There is no `leadingComments`...

Reply:
And location of `ParenthesizedExpression` seems not right,

```js
require('meriyah').parse('(x)', {preserveParens: true, ranges: true}).body[0].expression
{
  type: 'ParenthesizedExpression',
  expression: { type: 'Identifier', name: 'x', start: 1, end: 2, range: [ 1, 2 ] },
  start: 1,
  end: 3,
  range: [ 1, 3 ]
}
```

Reply:
I think we can try this later, and better make it work for all parsers.

Reply:
Okay.

BTW, `ParenthesizedExpression` nodes will be also very helpful if we decide to keep parens around subexpressions with comments. That would fix a couple of issues (e.g. #7746, #8809, probably #6945) and in general seems to be the right thing to do.

Reply:
Instead of relay on `ParenthesizedExpression`, I prefer implement something like this function https://eslint-utils.mysticatea.dev/api/ast-utils.html#isparenthesized

Reply:
Having separate nodes for parens makes everything easier, especially handling nested parens.